src/py42/clients/detectionlists/departing_employee.py

```diff
@@ -2,28 +2,37 @@
 
 from py42.clients import BaseClient
 from py42.clients.util import get_all_pages
+from py42.exceptions import Py42NotFoundError, Py42BadRequestError
 
 
 class DepartingEmployeeClient(BaseClient):
     """A client for interacting with Code42 Departing Employee APIs."""
 
     _uri_prefix = u"/svc/api/v2/departingemployee/{0}"
 
-    def __init__(self, session, user_context):
+    def __init__(self, session, user_context, detection_list_user_client, user_client):
         super(DepartingEmployeeClient, self).__init__(session)
         self._user_context = user_context
+        self._user_client = user_client
+        self._detection_list_user_client = detection_list_user_client
 
     def add(self, user_id, departure_date):
-        """Adds a user to Departing Employees.
+        """Adds a user to Departing Employees, creates the detection list profile if it doesn't
+        exist before adding.
         `REST Documentation <https://ecm-east.us.code42.com/svc/swagger/index.html?urls.primaryName=v2#/>`__
 
         Args:
-            user_id (str or int): Id of the departing employee.
+            user_id (str or int): The Code42 userUid of the user, to be added to departing employee.
             departure_date (date): Date in YYYY-MM-DD format
 
         Returns:
             :class:`py42.response.Py42Response`
         """
+        try:
```

Comment:
It would be better to move this code into `detection_list_user_client` in a method like `check_and_create` or something better name. As the code is identical in `HighRiskEmployeeClient` too.

By that we won't even need a direct dependency of `user_client` in `HighRiskEmployeeClient` or `DepartingEmployeeClient`.  
I can change it if you can suggest a appropriate method name for the same.

Reply:
I like this idea. How about `create_if_not_exists`?

Reply:
Changes done.